### PR TITLE
Discretization, Donor Cell and Central Differences

### PR DESCRIPTION
**Pull request by kimkroener**
added .cpp files for three files (see title), adapted relevant headers and makefile

### CODE REVIEW

File: src/discretization/1_discretization.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// Contructor
+Discretization::Discretization(std::array<int,2> nCells, std::array<double,2> meshWidth) :
+    StaggeredGrid(nCells, meshWidth)
+{
+}
+
+
+//! compute the 2nd derivative ∂^2 u / ∂x^2 at right edge
+double computeD2uDx2(int i, int j) const
```
Comment (MagnusOstertag) [inline src/discretization/1_discretization.cpp]: You have to add the namespace to all method calls like this
```suggestion
double Discretization::computeD2uDx2(int i, int j) const
```

File: src/discretization/1_discretization.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{
+    const double hx = this->meshWidth_[0] ; // mesh width in x direction
+
+    // return 2nd derivative ddu/dxx at right edge of cell i,j
+    return 1./(hx*hx) * (u(i-1,j) - 2.*u(i,j) + u(i-1,j));
```
Comment (MagnusOstertag) [inline src/discretization/1_discretization.cpp]: The members of the StaggeredGrid-instances are FieldVariable with an underscore behind the variable
```suggestion
    return 1./(hx*hx) * (u_(i-1,j) - 2.*u_(i,j) + u_(i-1,j));
```

**Review by MagnusOstertag — CHANGES_REQUESTED** [review]

File: src/discretization/2_central_differences.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include "discretization/1_discretization.h"
+
+//! compute the 1st derivative ∂ u^2 / ∂x
+double CentralDifferences::computeDu2Dx(int i, int j) const 
+{
+    const double uCenterRight = (u(i+1,j) + u(i,j))   / 2.; // u at the center of the right cell (i+1,j)
+    const double uCenterCell  = (u(i,j)   + u(i-1,j)) / 2.; // u at the center of the current cell (i,j)
+
+    // return central difference of squared values
+    return  (uCenterRight*uCenterRight - uCenterCell*uCenterCell)/hx();
```
Comment (MagnusOstertag) [inline src/discretization/2_central_differences.cpp:12]: dx()

File: src/discretization/1_discretization.cpp (hunk 1 above)
Comment (kimkroener) [reply src/discretization/1_discretization.cpp]: done

File: src/discretization/2_central_differences.cpp (hunk 3 above)
Comment (kimkroener) [reply src/discretization/2_central_differences.cpp:12]: done

**Pull request closed without merging**